### PR TITLE
CT-3385 attempt to fix 2 issues for db anon

### PR DESCRIPTION
**Pull request by ymao2**
## Description

This PR is to address 2 issues 
1 - Restoring the anonymised data from sql files into AWS RDS instance
2 - Introduce retry mechanism and printing out the error response to help when some errors happen during a long process of uploading sql files into bucket e.g. Connection to AWS bucket may drop (this is on my guess right now)

## Self-review checklist

* [ ] (1) Quick stakeholder demo done OR
* [ ] (2) ...bug with before and after screenshots
* [X] (3) Tests passing
* [X] (4) Branch ready to be merged (not work in progress)
* [X] (5) No superfluous changes in diff
* [X] (6) No TODO's without new ticket numbers
* [X] (7) PR Prefixed with ticket number e.g. `CT-7654 ...`

### Related JIRA tickets

https://dsdmoj.atlassian.net/secure/RapidBoard.jspa?rapidView=25&modal=detail&selectedIssue=CT-3385

### CODE REVIEW

File: lib/db/database_loader.rb [hunk 1]
```diff
@@ -20,12 +20,22 @@ def add_role(filename)
...
   end
 
+  # def drop_and_recreate_database
+  #   ENV['DISABLE_DATABASE_ENVIRONMENT_CHECK'] = '1'
+  #   Rake::Task['db:drop'].invoke
+  #   puts "Database dropped"
+  #   Rake::Task['db:create'].invoke
+  #   puts "Database created"
+  # end
+
```
Comment (Polsonby) [inline lib/db/database_loader.rb:30]: Do we still need this commented out version of the method?

File: lib/db/database_loader.rb [hunk 2]
```diff
@@ -20,12 +20,22 @@ def add_role(filename)
...
   def drop_and_recreate_database
+    # The following way for resetting the db has no need to stop the app server instances 
+    # but after the restoring, you may need to restart the instances in order to clear up the cache.
     ENV['DISABLE_DATABASE_ENVIRONMENT_CHECK'] = '1'
```
Comment (Polsonby) [inline lib/db/database_loader.rb:34]: Sorry, haven't taken the time to look into this, but is there any way this can be run against production?

Comment (ymao2) [reply lib/db/database_loader.rb:34]: No chance,  unless someone change my code as at very beginning,  I will check the env variable, if prod won't run, but you do have point, actually I don't need this line any more, as I don't drop/reset DB via rake task, let me remove it

File: lib/db/database_loader.rb (hunk 1 above)
Comment (ymao2) [reply lib/db/database_loader.rb:30]: Let me remove it

**Review by Polsonby — APPROVED** [review]
LGTM :) thanks for taking the time to keep iterating on this! 🙏

**Pull request merged**